### PR TITLE
Update permission on operator level

### PR DESCRIPTION
**Pull request by zdtsw**
## Description
ref: https://github.com/opendatahub-io/opendatahub-operator/issues/422
- to refine permission for operator instead of wildly open all group and all resources.
- removed DSCI and DSC editor and viewer permission in config, which are not used as input for CSV

also add MANIFEST_RELEASE to pass into image build if it is done by "make image" otherwise it is using v1.8 from odh-manifests

## How Has This Been Tested?

test: image: 'quay.io/wenzhou/opendatahub-operator:dev-2.8.15-40'

PS: if enable "ray" for the current ODH odh-manifests, we see reconcile from operator v2.
so we need to get https://github.com/opendatahub-io/odh-manifests/pull/910 into "master" odh-manifests to finalize the function. but from operator perspective, permissions are ready.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -94,41 +207,844 @@ rules:
...
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - route.openshift.io
+  resources:
+  - routes
+  verbs:
```
Comment (VaishnaviHire) [inline config/rbac/role.yaml:839]: Missing `patch` permission

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -94,41 +207,844 @@ rules:
...
+  resources:
+  - prometheusrules
+  verbs:
+  - create
+  - get
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
```
Comment (VaishnaviHire) [inline config/rbac/role.yaml:657]: Missing `patch` permission

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -94,41 +207,844 @@ rules:
...
+  - create
+  - delete
+  - list
+  - update
+  - watch
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - networkpolicies
+  verbs:
```
Comment (VaishnaviHire) [inline config/rbac/role.yaml:702]: Missing `patch` permission

File: config/rbac/role.yaml [hunk 4]
```diff
@@ -94,41 +207,844 @@ rules:
...
+  - create
+  - delete
+  - list
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - prometheuses
+  verbs:
+  - create
+  - get
```
Comment (VaishnaviHire) [inline config/rbac/role.yaml:645]: Missing `patch`

**Review by VaishnaviHire — COMMENTED** [review]
- Since we are doing serverside apply, resources should have `patch` permissions under `verbs`

- To successfully get deleted resources should also have `delete` permissions

File: config/rbac/role.yaml (hunk 4 above)
Comment (VaishnaviHire) [reply config/rbac/role.yaml:645]: Missing `delete`

**Comment by zdtsw** [comment]
latest test to set:
```
components:
    codeflare:
      enabled: true
    dashboard:
      enabled: false
    datasciencepipelines:
      enabled: true
    kserve:
      enabled: false
    modelmeshserving:
      enabled: false
    ray:
      enabled: false
    workbenches:
      enabled: true
```
get `DEBUG	events	DataScienceCluster instance default created and deployed successfully	{"type": "Normal", "object": {"kind":"DataScienceCluster","name":"default","uid":"20bb083c-c921-4d33-a9ce-0e3eb8308c00","apiVersion":"datasciencecluster.opendatahub.io/v1alpha1","resourceVersion":"8364229"}, "reason": "DataScienceClusterCreationSuccessful"}`

**Comment by zdtsw** [comment]
to solve  dashboard and modelmesh secret issue, we need `secrets/finalizers` to be added

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**